src/vs/editor/contrib/inlineCompletions/browser/singleTextEdit.ts

```diff
@@ -248,3 +247,58 @@ function smartDiff(originalValue: string, newValue: string, smartBracketMatching
 
 	return new LcsDiff({ getElements: () => elements1 }, { getElements: () => elements2 }).ComputeDiff(false).changes;
 }
+
+/**
+ * Given some text edits, this function finds the new ranges of the editted text post application of all edits.
+ * Assumes that the edit ranges are disjoint
+ * @param edits edits applied
+ * @returns new ranges post edits for every edit
+ */
+export function getNewRanges(edits: SingleTextEdit[]): Range[] {
+
+	if (edits.length === 0) {
+		return [];
+	}
+
+	const sortIndices = Array.from(edits.keys()).sort((a, b) =>
```

Comment:
I don't understand what you use `sortIndices` for.
You can just sort the edits array directly!

Reply:
So the reason I do this is because, from what I understand, when we set the selection with `editor.setSelections`, the first selection in the array will be the primary selection. In order to keep the primary selection the same selection as before, I would like the ranges post edits to be returned in the same order as the initial edits. Since the first edit is the primary edit, then the first range thus returned should be the primary edit range. 

The code is not needed otherwise.

Reply:
In general also I thought to myself, another user who uses this utility function later on, may expect the edit ranges to be returned in the same order as the edits.

Reply:
Ahh, you do this to maintain the final order!

I think I would break this up. This function should require that the edits are already sorted.
The caller has to solve this problem.

For that, you can have a helper like this:
```
function sortReversible<T>(array: readonly T[], compareFn: (a: T, b: T) => number): { sorted: T[], sortedIdxToOriginalIdx: (idx: number): number }

const r = sortReversible(possibleUnsortedEdits, mySortFn);
const sortedRanges = getNewRanges(r.sorted);
const ranges = sortedRanges.map((_, idx) => sortedRanges[r.sortedIdxToOriginalIdx(idx)])
```